shared/constants/teams.js

```diff
@@ -84,26 +85,11 @@ const getTopicFromConvID = (state: TypedState, conversationIDKey: ChatTypes.Conv
 const getParticipants = (state: TypedState, conversationIDKey: ChatTypes.ConversationIDKey) =>
   state.entities.getIn(['teams', 'convIDToChannelInfo', conversationIDKey, 'participants'], I.Set())
 
-const getMembersFromConvID = (state: TypedState, conversationIDKey: ChatTypes.ConversationIDKey) => {
-  const teamname = getTeamNameFromConvID(state, conversationIDKey)
-  if (teamname) {
-    return state.entities.teams.teamNameToMembers.get(teamname, I.Set())
-  }
-  return I.Set()
-}
-
-const getYourRoleFromConvID = (state: TypedState, conversationIDKey: ChatTypes.ConversationIDKey) => {
-  const members = getMembersFromConvID(state, conversationIDKey)
-  const you = usernameSelector(state)
-  const youAsMember = members.find(m => m.username === you)
-  if (youAsMember) {
-    return youAsMember.type
-  }
-  return null
-}
+const getRole = (state: TypedState, teamname: Types.Teamname) =>
+  state.entities.getIn(['teams', 'teamNameToRole', teamname], 'none')
 
-const isAdmin = (type: Types.TeamRoleType) => type === 'admin'
-const isOwner = (type: Types.TeamRoleType) => type === 'owner'
+const isAdmin = (type: ?string) => type === 'admin'
```

Comment:
Relaxed the typing here to avoid repeating `|| ''` everywhere and because the check still works

Reply:
sad that we're throwing away the type info, though. Perhaps typing `getRole` to return `TeamRole` or `?TeamRole` and then making this take `?TeamRole` would work?